### PR TITLE
Introduce Input component

### PR DESCRIPTION
**Pull request by dgalarza**
- Add jest-styled-components
- Port over Input component from BuoyRails

**Base**
image

**Disabled**
image

**With Error**
image

**With unit**
image

**Prefix unit**
image

### CODE REVIEW

File: packages/components/form/__specs__/input.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      const { getByTestId } = render(
+        <Input id="example" unit="$" prefixUnit={true} name="example" />
+      );
+
+      expect(getByTestId("input-unit")).toHaveStyleRule(
+        "margin-right",
+        theme.space.xs
+      );
+    });
+  });
```
Comment (acostanzo) [inline packages/components/form/__specs__/input.spec.tsx:59]: Is the prefix something we are intentionally carrying over as part of the new design system? I believe the only place they are currently used is the facility config which doesn't follow any of our designs.

**Review by acostanzo — APPROVED** [review]
One Q about unit prefixes but otherwise looks like a straightforward port

File: packages/components/form/__specs__/input.spec.tsx (hunk 1 above)
Comment (dgalarza) [reply packages/components/form/__specs__/input.spec.tsx:59]: Erik has it built into the new design system so it is intentional.

Comment (acostanzo) [reply packages/components/form/__specs__/input.spec.tsx:59]: Sweet!

**Pull request merged**